content/tutorials/2.projects/use-directus-templates-in-clay-for-data-enrichment.md

```diff
@@ -0,0 +1,417 @@
+
+### Sorting and Pagination
+
+**Sorting:**
+
+- **Key:** `sort`
+- **Value:** Field name (prefix with `-` for descending)
+- Example: `-date_created` (newest first)
+
+**Pagination:**
+
+- **Key:** `limit` - Maximum items to return (e.g., `50`)
+- **Key:** `offset` - Number of items to skip (e.g., `0`, `50`, `100`)
+
+---
+
+## Common Use Cases
+
+### E-commerce Product Enrichment
+
+**Scenario:** Enrich product data with inventory and pricing information
+
+1. Use **Get Item from Collection** to check if a product exists in Directus
+2. Use **Update Item in Collection** to update stock levels from external sources
+3. Use **Get Related Item Details** to pull supplier information
+
+### Content Publication Workflow
+
+**Scenario:** Automatically publish content when it's approved in Clay
+
+1. Use Directus webhooks to send draft content to Clay for review
+2. Enrich content with SEO metadata and keyword research in Clay
+3. Use **Update Item in Collection** to publish content back to Directus
+
+### Lead Enrichment System
+
+**Scenario:** Sync CRM data between Clay and Directus
+
+1. Use Directus webhooks to send new leads to Clay
+2. Enrich leads with company data and contact information
+3. Use **Update Item in Collection** to sync enriched data back
+4. Use **Get Related Item Details** to pull company profiles
+
+### Form Submission Processing
+
+**Scenario:** Process form submissions and create records
+
+1. Use Directus webhooks to send form submissions to Clay
+2. Use **Get Item from Collection** to check for existing records
+3. Use **Create Item in Collection** to add new contacts
+4. Use **Update Item in Collection** to update existing records
+
+---
+
+## Best Practices
+
+### Authentication and Security
+
+- Use API tokens with minimal required permissions (follow the principle of least privilege)
+- Never share tokens in screenshots or public documentation
+- Regularly rotate your authentication credentials
+- Use separate tokens for different environments (development, staging, production)
+
+### Performance Optimization
+
+- **Select only needed fields** - Use the `fields` parameter to reduce data transfer
+- **Implement rate limiting** - Avoid hitting API limits with large batch operations
+- **Use filters effectively** - Narrow results at the API level rather than in Clay
+- **Batch operations carefully** - Break large operations into smaller chunks
+
+### Error Handling
+
+- Set up proper error handling for failed requests in Clay
+- Monitor enrichment success rates regularly
+- Have fallback strategies for critical data operations
+- Test thoroughly with sample data before running on production
+
+### Data Mapping
+
+- Use intermediary formula columns for complex transformations
+- Validate data types before sending to Directus
+- Handle optional fields gracefully (don't send empty values if not needed)
+- Keep track of your schema changes and update templates accordingly
+
+---
+
+## Troubleshooting
```

Comment:
Do we need to list all of the generic directus errors the api responds with, or can we just link to them?